### PR TITLE
accept integer and float columns in wide mode

### LINKED ISSUE #2561: wide-mode type checking too strict
As reported in https://community.plotly.com/t/df-plot-kind-box-with-plotly-returns-error-of-wide-form-data/40729/5?u=nicolaskruchten it would be nice if a mix of floats and ints were still accepted for wide-mode processing, rather than strictly saying the types don't match.

### PR DESCRIPTION
**Pull request by nicolaskruchten**
Closes #2561

## Code PR

- [x] I have added tests (if submitting a new feature or correcting a bug) or
  modified existing tests.
- [x] I have added a CHANGELOG entry if fixing/changing/adding anything substantial.

### CODE REVIEW

File: packages/python/plotly/plotly/express/_core.py [hunk 1]
```diff
@@ -1390,9 +1390,11 @@ def build_dataframe(args, constructor):
         del args["wide_cross"]
         dtype = None
         for v in wide_value_vars:
+            v_dtype = df_output[v].dtype.kind
+            v_dtype = "number" if v_dtype in ["i", "f"] else v_dtype
```
Comment (emmanuelle) [inline packages/python/plotly/plotly/express/_core.py:1394]:
```suggestion
            v_dtype = "number" if v_dtype in ["i", "f", "u"] else v_dtype
```

Comment (emmanuelle) [reply packages/python/plotly/plotly/express/_core.py:1394]: "u" is for unsigned integer it's used in images for example

**Comment by emmanuelle** [comment]
:woman_dancing: (I suggested to add `"u"` in the number types)

**Comment by nicolaskruchten** [comment]
Awesome, thanks for catching that... I'm not as familiar with the types here as I should be :)

**Pull request merged**